### DISCUSSION TITLE
How can I convert these settings to filter argument?

### DISCUSSION THREAD

**Discussion by f1refa11** [miurahr/py7zr#560] [Q&A]
I use these settings in 7-zip to make an archive:
![изображение](https://user-images.githubusercontent.com/58619409/222047173-ac5b13ab-4d71-4bb4-a19a-d05b35c65033.png)
How can I convert these settings to `filter` argument in `py7zr.SevenZipFile`?

**Comment by miurahr** [comment]
from 7-zip document, a compression level 9 uses LZMA2 and BCJ2
https://documentation.help/7-Zip/method.htm#SevenZipX

Python core library and py7zr does not support BCJ2.
https://github.com/miurahr/py7zr/#not-supported-algorithms

**Comment by miurahr** [comment]
I'd like to advice you to check also a standard python document for filter values.
https://docs.python.org/3/library/lzma.html#specifying-custom-filter-chains

py7zr pass a specified filter value to python lzma module.

**Comment by f1refa11** [comment]
> I'd like to advice you to check also a standard python document for filter values. https://docs.python.org/3/library/lzma.html#specifying-custom-filter-chains
> 
> py7zr pass a specified filter value to python lzma module.

Thanks, but I don't see solid block size value support, does lzma module support setting block size to Solid?

**Comment by miurahr** [comment]
`solid` compression is not a filter settings, but a way to call write functions.

You can open `SevenZipFile` object and calls multiples of `write`  and close 
 a object, when you can observe a solid compression archive.

https://py7zr.readthedocs.io/en/latest/api.html#py7zr.SevenZipFile.write

And you can open the file with `append` mode and `write` again,
you will observe a two solid blocks.

py7zr does not provide a native way to handle solid blocks in single opened object.
